### ISSUE TITLE
Can not update to 1.55.0: ValueError: Empty filename

### ISSUE THREAD

**Issue by psi-4ward** [SukramJ/hahomematic#1423]
### I agree to the following

- [X] I have read the [documentation](https://github.com/danielperna84/custom_homematic#custom_homematic) right now
- [X] I have read the [FAQs](https://github.com/danielperna84/custom_homematic#frequently-asked-questions) right now
- [X] I am aware of the latest [release notes](https://github.com/danielperna84/custom_homematic/releases)
- [X] I am aware that an integration that was installed via HACS also needs to be updated via HACS
- [X] The Backend (CCU/Homegear/...) is working as expected, devices are taught in and are controllable by its own UI.
- [X] I am running the latest version of the custom_component (and Home Assistant)

### The problem

I've just clicked the update button in HA and then I get an Error: `ValueError: Empty filename.`

```
INFO (MainThread) [custom_components.hacs] <Integration danielperna84/custom_homematic> Trying experimental download
DEBUG (MainThread) [custom_components.hacs] Downloading https://api.github.com/repos/danielperna84/custom_homematic/zipball/1.55.0

ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [139729245408064] Error handling message: Unknown error (unknown_error) Psi from ... (Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/decorators.py", line 26, in _handle_async_response
    await func(hass, connection, msg)
  File "/config/custom_components/hacs/websocket/repository.py", line 234, in hacs_repository_download
    await repository.async_install()
  File "/config/custom_components/hacs/repositories/base.py", line 899, in async_install
    await self.async_install_repository()
  File "/config/custom_components/hacs/repositories/base.py", line 982, in async_install_repository
    await self.download_content()
  File "/config/custom_components/hacs/repositories/base.py", line 629, in download_content
    await self.download_repository_zip()
  File "/config/custom_components/hacs/repositories/base.py", line 686, in download_repository_zip
    zip_file.extractall(self.content.path.local, extractable)
  File "/usr/local/lib/python3.12/zipfile/__init__.py", line 1721, in extractall
    self._extract_member(zipinfo, path, pwd)
  File "/usr/local/lib/python3.12/zipfile/__init__.py", line 1762, in _extract_member
    raise ValueError("Empty filename.")
ValueError: Empty filename.
```

Probably a problem with HACS?

### What version of HomematicIP (local) has the issue?

1.55.0

### What was the last working version of HomematicIP (local)?

1.54.0

### What type of installation are you running?

Home Assistant Supervised

### What type of installation are you running for your homematic backend?

RaspberryMatic Standalone

### Which config details do you use

- [ ] TLS
- [ ] callback data ([see](https://github.com/danielperna84/custom_homematic#callback_host-and-callback_port))

### Which interfaces do you use?

- [ ] Homematic IP
- [ ] Homematic (Bidcos-RF)
- [ ] Groups (Heating-Group)
- [ ] BidCos-Wired (HM-Wired)

**Comment by SukramJ** [comment]
What is the update button in HA? Do you mean HACS?
Updating/Downloading via HACS is not part of the integration.
Please report it to HACS.

**Comment by psi-4ward** [comment]
![image](https://github.com/danielperna84/hahomematic/assets/1191572/f09e3d21-f0d2-4a54-86f9-7d57c0e3aeb1)

This one.

**Comment by SukramJ** [comment]
Please report it to HACS.

**Comment by psi-4ward** [comment]
https://github.com/hacs/integration/issues/3465

https://github.com/hacs/integration/pull/3456/files

## TLDR

When upgrading python to 3.12 and having HACS v 1.33 installed you can not extract the zipball download from github.

After updating HACS to v1.34 the homematic_local can be updated.
